addons/mail/static/src/core/common/message.xml

```diff
@@ -148,15 +148,27 @@
             }"
         >
             <t t-foreach="messageActions.actions.slice(0, quickActionCount)" t-as="action" t-key="action.id">
+                <t t-set="firstAction" t-value="action.id === messageActions.actions[0].id"/>
+                <t t-set="lastAction" t-value="messageActions.actions.length gt quickActionCount and action.id === messageActions.actions[messageActions.actions.length - 1].id"/>
                 <t t-if="action.callComponent" t-component="action.callComponent" t-props="action.props"/>
-                <button t-else="" class="btn px-1 py-0 rounded-0" t-att-title="action.title" t-att-name="action.id" t-on-click.stop="action.onClick">
+                <button
+                    t-else=""
+                    class="btn px-1 py-0 rounded-0"
+                    t-att-class="{
+                        'rounded-start-1': firstAction,
+                        'rounded-end-1': lastAction,
+                    }"
+                    t-att-title="action.title"
+                    t-att-name="action.id"
+                    t-on-click.stop="action.onClick"
+                >
```

Comment:
We know XML sucks, but let's not making newlines for each attribute.
This makes seeing details better, but it's gives a much harder time to have a macro-view of the template.
We value far more the macro-view of template, thus we prefer to inline most things even if this is uglier.

It's fine in a few selected cases when a few attributes are long, e.g. with `t-att-class` object syntax and you have at least 2 conditionals. Only make newline on the specific attributes rather than all attributes of the node.

```xml
<button t-else="" class="btn px-1 py-0 rounded-0" t-att-title="action.title" t-att-name="action.id" t-on-click.stop="action.onClick" t-att-class="{
    'rounded-start-1': firstAction,
    'rounded-end-1': lastAction,
}">
```